docs/layout-of-source-files.rst

```diff
@@ -216,101 +216,62 @@ the code below creates new global symbols ``alias`` and ``symbol2`` which refere
 
   import {symbol1 as alias, symbol2} from "filename";
 
-Paths
------
+.. index:: virtual filesystem, source unit name, import; path, filesystem path
 
-In the above, ``filename`` is always treated as a path with ``/`` as directory separator,
-and ``.`` as the current and ``..`` as the parent directory.  When ``.`` or ``..`` is followed by a character except ``/``,
-it is not considered as the current or the parent directory.
-All path names are treated as absolute paths unless they start with the current ``.`` or the parent directory ``..``.
+Import paths
+------------
 
-To import a file ``filename`` from the same directory as the current file, use ``import "./filename" as symbolName;``.
-If you use ``import "filename" as symbolName;`` instead, a different file could be referenced
-(in a global "include directory").
+The paths used in imports in a general case do not have to be filesystem paths.
+The compiler maintains an internal database (:ref:`virtual filesystem <virtual-filesystem>`) where
+each compiled source unit is assigned a unique *source unit name* which is an opaque and unstructured identifier.
+The import path is translated into a source unit name and used to find the corresponding source unit
+in this database.
+If you directly provide source code to the compiler using the :ref:`Standard JSON interface
+<compiler-api>`, you can choose arbitrary source unit names and use them in your imports, completely
+ignoring the actual filesystem.
 
-It depends on the compiler (see :ref:`import-compiler`) how to actually resolve the paths.
-In general, the directory hierarchy does not need to strictly map onto your local
-filesystem, and the path can also map to resources such as ipfs, http or git.
+While this mechanism provides a lot of freedom, in most cases the source files do reside on disk
+and it is convenient if the compiler can find and load them automatically.
+For this reason, if the source unit name corresponding to the import path is not found in the
+virtual filesystem, the compiler invokes the file loader.
+If you are using the command-line interface, the file loader simply assumes that the source unit
+name passed to it is a path and tries to load the file from the actual filesystem.
+The `JavaScript interface <https://github.com/ethereum/solc-js>`_ is a bit more flexible in that
+regard and allows the user to provide a callback to load the file.
+`The Remix IDE <https://remix.ethereum.org/>`_ uses this mechanism to allow files to be imported
+directly from github.
 
-.. note::
-    Always use relative imports like ``import "./filename.sol";`` and avoid
```

Comment:
I would prefer this to still be in a prominent place.

Reply:
Done.